src/Authentication/Authentication.Core/Interfaces/IGraphRequestProofofPossession.cs

```diff
@@ -0,0 +1,22 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using Azure.Core;
+using Azure.Identity;
+using System;
+using System.Net.Http;
+
+namespace Microsoft.Graph.PowerShell.Authentication
+{
+    public interface IGraphRequestProofofPossession
```

Comment:
This public interface is confusing me. There are no comments. It combines several things in it: 

- SHR details (uri, method) - part of the request
- access token - part of the response
- Azure Identity concept "credential" 
- Graph concepts.

I tend to think this is more of a https://en.wikipedia.org/wiki/God_object